docs/source/howto/query.rst

```diff
@@ -27,6 +27,13 @@ Both APIs provide the same functionality - the appender method may be more suita
 In this section we will focus on the basics of the appender method.
 For more advanced queries or more details on the query dictionary, see the :ref:`topics section on advanced querying <topics:database:advancedquery>`.
 
+.. admonition:: New in version 2.6
+    :class: aiida-versionadded
+
+    We've added a new syntax for defining filters and projections in the :class:`~aiida.orm.querybuilder.QueryBuilder` that provides a user-friendly API for query construction.
+    Note that this new syntax can be mixed together with the traditional syntax in the same query.
+    Keep an eye out for similar **"New"** blocks for further information regarding the new syntax.
```

Comment:
Instead of this vague forward reference, it would perhaps be better to provide a direct clickable link